_posts/2018-03-23-minnowboard-turbot-remote-firmware-flashing-with-remote-testing-environment.md

```diff
@@ -0,0 +1,312 @@
+connector. Then DC Jack - DC Jack wire was connected to the `J13` of RTE and
+to the `J9` of Minnowboard. That configuration allows to control power supply
+remotely.
+
+* SPI
+
+| Minnowboard J1 | Color         | RTE                      |
+|:--------------:|:-------------:|:------------------------:|
+| 1              | red           | 1 (J9)                   |
+| 2              | black         | 2 (J1)                   |
+| 3              | orange        | 3 (J1) + 1.2 kΩ resistor |
+| 4              | green         | 4 (J1)                   |
+| 5              | blue          | 5 (J1)                   |
+| 6              | brown         | 6 (J1)                   |
+| 7              | -             | NC                       |
+| 8              | -             | NC                       |
+
+* UART debug
+
+| Minnowboard J4 | Color         | UART-USB converter |
+|:--------------:|:-------------:|:------------------:|
+| 1              | black         | GND                |
+| 2              | -             | NC                 |
+| 3              | -             | NC                 |
+| 4              | blue          | TX                 |
+| 5              | yellow        | RX                 |
+| 6              | -             | NC                 |
+
+
+There was one but important problem with the built system. Memory flashing was
+realized correctly, but Minnowboard doesn't boot. The source of the problem 
+was `CS#` signal. It was set to `0`, so memory was always enabled, but it was 
+not enough. Minnowboard still doesn't boot. So I decided to use an 
+oscilloscope. That analysis showed me that Minnowboard Turbot doesn't set 
+memory chip's `CS#` input to `0` all the time. It chnges over the time. Mostly
+`CS#` is set to `0`, but periodically it's being set to `1`.
+
+Time analysis of the `CS#` states seemed to be too time-consuming and trying
+correct control of the `CS#` lines too inflexible in the event of changes.
+I needed to reduce the impact of the `CS#` RTE output to make it a Minnowboard
+Turbot took precedence in choosing the state of the `CS#` line.
+I decided to use 1.2 kΩ resistor between RTE `CS#` output and Minnowboard 
+Turbot `CS#` input and this was a good idea.
+
+After resolving `CS#` problem, flashing procedure is very simple.
+It amounts to: 
+1. Turn off Minnowboard platform power supply.
+2. Flash Minnowboard ROM memory via SPI.
+3. Turn on Minnowboard platform power supply.
+
+And it's that all.
+
+### Firmware flashing
+
+I used 2 firmwares to test presented Remote Testing Environment:
+
+[MNW2MAX1.X64.0097.D01.1709211100.bin](https://firmware.intel.com/sites/default/files/MinnowBoard_MAX-0.96-Firmware.Images.zip),
+
+[MNW2MAX1.X64.0097.D01.1709211100.bin](https://firmware.intel.com/sites/default/files/MinnowBoard_MAX-Rel_0_97-Firmware.Images.zip).
+
+
+Code of flashing script:
+```
+#!/bin/bash
+
+# turn off Minnowboard platform power supply
+../rte_ctrl -rel
```

Comment:
@acihy this component is closed source. We have to release this code under BSD license I think. @mek-x any ideas here?

@acihy we should not publish a blog post with closed component since no one can reproduce that.